docs/api.md

```diff
@@ -148,6 +149,9 @@ Types used by the former:
    experimental.ReadCallback
    experimental.WriteCallback
    experimental.StorageType
+   experimental.backed._lazy_arrays.MaskedArray
+   experimental.backed._lazy_arrays.CategoricalArray
+   experimental.backed._xarray.Dataset2D
```

Comment:
Hmmm, that's an interesting point.  I thought that by exporting `read_lazy` in general, I might appease @ivirshup pre-emptively by not creating a monolith function and making something more composable a la `read_elem`.  But in fact, by doing that, we create more API surface, yes.  `Dataset2D` is something that will be visible on the `AnnData` object via `adata.obs` but the others would not be if we didn't make `read_lazy` so general (i.e., because we added functionality to it from its previous state as `read_elem_as_dask`).  Tough question.

Reply:
As said in the “X type consolidation” PR, I think it’s fine to not export the full API surface of the classes we have and just commit to a carefully designed protocol instead.

Reply:
If we do this, we don’t need to write anything, people will only see the superclass, i.e. the stable API:

```py
qualname_overrides = {
    ...,
    "anndata.xyz.MaskedArray": "numpy.ma.Array",
}
```

Reply:
Back to our sphinx-autodoc-singledispatch issue, I don't think we can do this because we don't install the package unless this can work without the package installed or mocked

Reply:
hmm, right the intersphinx mapping is independent of installed packages!

Reply:
Ok @flying-sheep I am not sure `qualname_overrides` works this way - I tried adding it in but could not get the links to redirect to the xarray docs